### PR TITLE
Floating dialog workflow

### PR DESCRIPTION
**Pull request by alan-wu**

### CODE REVIEW

File: src/components/MapPopover.vue [hunk 1]
```diff
@@ -56,37 +56,56 @@ export default {
...
-              type: "Plot",
-              plotType: "heatmap",
-            }
-          ];
-        } else {
-          data.title = term;
-          data.description = "";
-          data.actions = [ ];
+        switch (term) {
+          case "UBERON:0000948":
```
Comment (Tehsurfer) [inline src/components/MapPopover.vue:60]: Might be good to compile these into a separate file where we can manage them if we get any more.

File: src/components/FloatingFlow.vue [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+      newEntry.type = data.type;
+      newEntry.mode = "normal";
+      if (data.plotType)
+        newEntry.plotType = data.plotType;
+      newEntry.id = ++this.currentCount;
+      newEntry.zIndex = ++this.zIndex;
+      this.entries.push(newEntry);
+      return newEntry.id;
+    },
+    findIndexOfId: function(id) {
```
Comment (Tehsurfer) [inline src/components/FloatingFlow.vue:62]: Can the position in entries array not be their 'id'?

File: src/components/FloatingFlow.vue [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+      */
+      let index = this.findIndexOfId(id);
+      if ((index > -1) && (this.entries[index].mode === "normal")) {
+        if (this.zIndex !== this.entries[index].zIndex) {
+          this.zIndex++;
+          this.entries[index].zIndex = this.zIndex;
+        }
+      }
+    },
+    distroyDialog: function(id) {
```
Comment (Tehsurfer) [inline src/components/FloatingFlow.vue:91]: 'distroy' to 'destroy'

File: src/components/FloatingFlow.vue [hunk 4]
```diff
@@ -0,0 +1,202 @@
+<template>
+  <el-container style="height:100%;background:white;">
+    <el-header style="text-align: left; font-size: 14px;padding:0" height="40px" class="dialog-header">
+      <DialogToolbarContent :activeId="activeDockedId" :dialogTitles="dockedArray"
+        :showIcons="entries[findIndexOfId(activeDockedId)].mode!=='main'" @maximise="dockedMaximise"
```
Comment (Tehsurfer) [inline src/components/FloatingFlow.vue:5]: So icons need to be hidden based on which dialog is the main one?

File: src/components/MapContent.vue [hunk 5]
```diff
@@ -1,212 +1,20 @@
...
-        <el-main class="dialog-main">
-          <div style="width:100%;height:100%;position:relative;overflow:hidden;">
-            <FloatingDialog v-for="item in entries" :entry="item" :index="item.id"
-              :key="item.id" v-on:mousedown.native="dialogClicked(item.id)"
-              @maximise="dialogMaximise(item.id)" @minimise="dialogMinimise(item.id)" 
-              @close="dialogClose(item.id)"/>
-          </div>
-        </el-main>
-      </el-container>
+      <FloatingFlow />
```
Comment (Tehsurfer) [inline src/components/MapContent.vue:3]: Does MapContent.vue do anything now that the Dialogs are removed?

File: src/components/MapPopover.vue (hunk 1 above)
Comment (alan-wu) [reply src/components/MapPopover.vue:60]: Thats correct, this should really be hooked into a scicrunch search backend.

File: src/components/FloatingFlow.vue (hunk 2 above)
Comment (alan-wu) [reply src/components/FloatingFlow.vue:62]: Not in this case, id is unique while posiiton changed when an item is removed.

File: src/components/FloatingFlow.vue (hunk 3 above)
Comment (alan-wu) [reply src/components/FloatingFlow.vue:91]: Ok, will fix that.

File: src/components/FloatingFlow.vue (hunk 4 above)
Comment (alan-wu) [reply src/components/FloatingFlow.vue:5]: Yes, the flatmap dialog can never be destroyed.

File: src/components/MapContent.vue (hunk 5 above)
Comment (alan-wu) [reply src/components/MapContent.vue:3]: It will be used to contain different workflow/prototype.

Comment (alan-wu) [reply src/components/MapContent.vue:3]: The idea is that MapContent should contain the information of the array of viewers which will then get passed into different workflows. (Allowing toggling of different workflows.)

Comment (alan-wu) [reply src/components/MapContent.vue:3]: We are not there yet but we will be.

File: src/components/MapPopover.vue (hunk 1 above)
Comment (Tehsurfer) [reply src/components/MapPopover.vue:60]: Yep I see

File: src/components/MapContent.vue (hunk 5 above)
Comment (Tehsurfer) [reply src/components/MapContent.vue:3]: Yep I see so you have a premade workflow and pas that to MapContent

File: src/components/FloatingFlow.vue (hunk 4 above)
Comment (Tehsurfer) [reply src/components/FloatingFlow.vue:5]: Ok so icons here relate to display of the flatmap dialog?

File: src/components/MapPopover.vue (hunk 1 above)
Comment (Tehsurfer) [reply src/components/MapPopover.vue:60]: Something like the equivalent of the 'augmented-sci-crunch-results' but on sci crunch ideally right?

File: src/components/FloatingFlow.vue (hunk 2 above)
Comment (Tehsurfer) [reply src/components/FloatingFlow.vue:62]: I'm still not sure I get it,  but I saw somewhere in a different file along the lines of `id == -1`. So I can see how it is used as more than just an index in an array.

File: src/components/MapPopover.vue (hunk 1 above)
Comment (alan-wu) [reply src/components/MapPopover.vue:60]: Yes, thats correct,

File: src/components/FloatingFlow.vue (hunk 2 above)
Comment (alan-wu) [reply src/components/FloatingFlow.vue:62]: It is just a way for me to make sure each entry has an identifier that is not going to be changed through out its life time.

File: src/components/FloatingFlow.vue (hunk 4 above)
Comment (alan-wu) [reply src/components/FloatingFlow.vue:5]: Yes, when mode is "main" means that it is the main dialog, should and cannot not be destroyed.

**Pull request merged**